client/jetpack-connect/main-wrapper.jsx

```diff
@@ -40,10 +40,13 @@ export class JetpackConnectMainWrapper extends PureComponent {
 			'is-mobile-app-flow': !! retrieveMobileRedirect(),
 		} );
 
+		const partnerSlug = isWoo ? 'woocommerce' : this.props.partnerSlug;
```

Comment:
I felt like sticking to data flow where you'd use either only `isWoo=true` _or_ `parterSlug='woocommerce` would be more clear and less surprising. That said, I understand this might be a constraint you're having with `JetpackHeader` that was introduced in the codebase already before this PR.

This definitely isn't blocking comment but are there cons on using `partnerSlug` across all these components?

Reply:
I can't think of any negative side-effects code wise.

Though one of the reasons I went with the `isWoo` pattern was from the existing separation I already saw here: https://github.com/Automattic/wp-calypso/blob/9b32491717f9f76a37c8fbc26572f22f6ccaf294/client/jetpack-connect/authorize.js#L230-L235, and it felt more future proof to me. There have been discussions about making it possible for WooCommerce / WooCommerce Services to just bundle authentication/sync modules and not rely on the full Jetpack plugin. It feels more right to consider this a different 'flow' to me than just one of the branded partner flows, since some of that branding could actually be dropped in the future. It also doesn't feel like WooCommerce is "partner" in the same way the hosts are, though I know that may be a bit pedantic.

Reply:
Another consideration is the opportunity for us in the future to co-brand this version of onboarding as well (I don't believe we have plans for this yet, but it's been raised). Having both properties would be useful -- so I can actually continue the `isWoo` pattern in `JetpackHeader` too.

Reply:
In https://github.com/Automattic/wp-calypso/pull/33073/commits/ec4b2b0da4a0c1137a0b4760fb87999cff3137aa, I implemented a `isWoo` parameter to keep this consistent and for the reasons above. I also added a `@todo` for handling mixing of `isWoo` and `partnerSlug` for further co-branded flows, but this is not currently in the scope of the project.

Reply:
Sounds great and totally valid reasoning :thumbsup: Thanks for giving it such an in-depth consideration!